include/sgm_stereo/sgm_stereo_node.h

```diff
@@ -0,0 +1,139 @@
+/*
+    Copyright (C) 2018  Teyvonia Thomas
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef SGM_STEREO_SGM_STEREO_NODE_H
+#define SGM_STEREO_SGM_STEREO_NODE_H
+
+// C++ Includes
+#include <string>
+#include <vector>
+#include <cstdint>
+#include <iostream>
+#include <cstring>
+#include <stack>
+#include <ctime>
+
+// ROS Includes
+#include <ros/ros.h>
+#include <ros/topic.h>
+#include <message_filters/synchronizer.h>
+#include <message_filters/sync_policies/approximate_time.h>
+#include <message_filters/subscriber.h>
+#include <cv_bridge/cv_bridge.h>
+#include <image_transport/image_transport.h>
+#include <sensor_msgs/image_encodings.h>
+#include <image_geometry/stereo_camera_model.h>
+#include <sensor_msgs/PointCloud2.h>
+#include <sensor_msgs/CameraInfo.h>
+#include <stereo_msgs/DisparityImage.h>
+#include <sensor_msgs/point_cloud2_iterator.h>
+
+// OpenCV Includes
+#include <opencv2/core/core.hpp>
+#include <opencv2/opencv.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <opencv2/highgui/highgui.hpp>
+
+#include "nodelet/nodelet.h"
+#include "sgm_stereo/SGMStereo.h"
+
+using std::vector;
+
+// Just a timer utility
```

Comment:
Move the tic toc utility into the implementation... or try to use something else more "standard"